xbmc/peripherals/devices/PeripheralJoystick.cpp

```diff
@@ -70,10 +70,18 @@ bool CPeripheralJoystick::InitialiseFeature(const PeripheralFeature feature)
   {
     if (feature == FEATURE_JOYSTICK)
     {
-      if (m_bus->InitializeProperties(*this))
-        bSuccess = true;
+      // Ensure an add-on is present to translate input
+      if (!m_manager.GetAddonWithButtonMap(this))
+      {
+        CLog::Log(LOGERROR, "CPeripheralJoystick: No button mapping add-on for %", m_strLocation.c_str());
```

Comment:
This or the next CLog are too early, CLog does not exist at the moment this Log is called -> segfault on shield during startup

Reply:
@FernetMenta is CLog a global we can kill?